src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -269,10 +271,104 @@ public void Stop()
             });
         }
 
+        /// <summary>
+        /// Gets a list of document IDs that are going to be pushed, but have not been pushed yet
+        /// <item type="bullet">
+        /// <description>API is a snapshot and results may change between the time the call was made and the time</description>
+        /// </item>
+        /// </summary>
+        /// <returns>An immutable set of strings, each of which is a document ID</returns>
+        /// <exception cref="CouchbaseLiteException">Thrown if no push replication</exception>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        [NotNull]
+        public IImmutableSet<string> GetPendingDocumentIDs()
+        {
+            var result = new HashSet<string>();
+            if (!IsPushing()) {
+                CBDebug.LogAndThrow(WriteLog.To.Sync,
+                    new CouchbaseLiteException(C4ErrorCode.Unsupported, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs),
+                    Tag, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs, true);
+            }
+
+            DispatchQueue.DispatchSync(() => {
+                var errSetupRepl = SetupC4Replicator();
+                if (errSetupRepl.code > 0) {
+                    CBDebug.LogAndThrow(WriteLog.To.Sync, CouchbaseException.Create(errSetupRepl), Tag, errSetupRepl.ToString(), true);
+                }
+            });
+
+            C4Error err = new C4Error();
+            byte[] pendingDocIds = null;
+            pendingDocIds = Native.c4repl_getPendingDocIDs(_repl, &err);
+
+            if (err.code > 0) {
+                CBDebug.LogAndThrow(WriteLog.To.Sync, CouchbaseException.Create(err), Tag, err.ToString(), true);
```

Comment:
Shouldn't this be an exception?

Reply:
Duh....I read it wrong....but use `LiteCoreBridge` for this purpose.  It saves lines.